mmaction/core/runner/omnisource_runner.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) Open-MMLab. All rights reserved.
+import time
+import warnings
+
+import mmcv
+from mmcv.runner import EpochBasedRunner, Hook
+from mmcv.runner.utils import get_host_info
+
+
+def cycle(iterable):
+    iterator = iter(iterable)
+    while True:
+        try:
+            yield next(iterator)
+        except StopIteration:
+            iterator = iter(iterable)
+
+
+class OmniSourceDistSamplerSeedHook(Hook):
+
+    def before_epoch(self, runner):
+        for data_loader in runner.data_loaders:
+            if hasattr(data_loader.sampler, 'set_epoch'):
+                # in case the data loader uses `SequentialSampler` in Pytorch
+                data_loader.sampler.set_epoch(runner.epoch)
+            elif hasattr(data_loader.batch_sampler.sampler, 'set_epoch'):
+                # batch sampler in pytorch wraps the sampler as its attributes.
+                data_loader.batch_sampler.sampler.set_epoch(runner.epoch)
+
+
```

Comment:
@RUNNERS.register_module()

Reply:
This class doesn't need registration, since we directly use it without calling any build_xxx function.